### DISCUSSION TITLE
Hey all , I'm in the process to migrate azure function from .net 3.1 to .net 7,but not able to see any support for [DurableClient] IDurableClient , any suggestions how to fix it?

### DISCUSSION THREAD

**Discussion by harmeetsingh170394** [Azure/azure-functions-durable-extension#2655] [Q&A]
` public async Task<IActionResult> Run(
            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "{resourceName}/myapi/updatexyz")] HttpRequest httpRequest,
             [DurableClient] IDurableClient durableClient)`

**Comment by empowerJustinVocke** [comment]
What version of Durable Functions and Azure Functions are you using? Is it In-Process or Isolated Worker?

**Reply by harmeetsingh170394** [reply]
<TargetFramework>net7.0</TargetFramework>
		<AzureFunctionsVersion>v4</AzureFunctionsVersion>

   "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated",

<img width="332" alt="image" src="https://github.com/Azure/azure-functions-durable-extension/assets/143993882/1ada2562-58da-44cd-b539-f5c866b8f0c8">

This is the version I'm using @empowerJustinVocke

**Reply by marked23** [reply]
When working in dotnet-**isolated** mode, your DF related nuget packages should have the word `Worker` in the namespace.  
If you have packages with the word `WebJobs` in the namespace, then those packages are probably not appropriate. 

In your function signature, change `IDurableClient` to `DurableTaskClient`

example:
```csharp
public async Task<HttpResponseData> MyProcess_HttpStart(
    [HttpTrigger]   HttpRequestData   req,
    [DurableClient] DurableTaskClient client,
                    FunctionContext   executionContext)
```

**Comment by ltouro** [comment]
try 
```
dotnet add package Microsoft.Azure.WebJobs.Extensions --version 5.0.0
```
```
using Microsoft.Azure.WebJobs.Extensions.DurableTask;
```
